rfcs/proposed/task_group_dynamic_dependencies/README.md

```diff
@@ -0,0 +1,365 @@
+clarification on these proposed extensions.
+
+1. Extend semantics and useful lifetime of `task_handle`. We propose `task_handle` 
+to represent tasks for the purpose of adding dependencies. The useful lifetime and 
+semantics of `task_handle` will need to be extended to include tasks that have been 
+submitted, are currently executing, or have been completed.
+2. Add functions to set task dependencies. In the current `task_group`, tasks can 
+only be waited for as a group and there is no direct way to add any before-after 
+relationships between individual tasks. We will discuss options for spelling.
+3. Add a function to move successors from a currently executing task to a new task. 
+This functionality is necessary for recursively generated task graphs. This case 
+represents a situation where it is safe to modify dependencies for an already 
+submitted task. 
+
+### Extend the semantics and useful lifetime of task_handle
+
+Dynamic tasks graphs order the execution of tasks via dependencies on the 
+completion of other tasks. They are dynamic because the creation of tasks, 
+specification of dependencies between, submission of tasks for scheduling, and
+execution of the tasks may happen concurrently and in various orders. Different
+concrete use cases have different requirements on when new tasks are created 
+and when dependencies between tasks are specified. 
+
+For the sake of discussion, let’s label four points in a task’s lifetime: 
+
+1. created
+2. submitted
+3. executing
+4. completed
+
+A created task has been allocated but is not yet known to the scheduling
+algorithm and so cannot begin executing. A submitted task is known to the 
+scheduling algorithm and whenever its incoming dependencies (predecessor tasks)
+are complete it may be scheduled for execution. An executing task has started 
+executing its body but is not yet complete. Finally, a completed task has
+executed fully to completion.
+
+In the current specification for `task_group`, the function `task_group::defer`
+already provides a mechanism to separate task creation from submission. 
+`task_group::defer` returns a `tbb::task_handle`, which represents a created 
+task. A created task is in the created state until it is submitted via
+the `task_group::run` or `task_group::run_and_wait` functions. In the current 
+specification of `task_group`, accessing a `task_handle` after it is submitted
+via one of the run functions is undefined behavior. Currently, therefore, a 
+`task_handle` can only represent a created task. And currently, any task
+that is run can immediately be scheduled for execution since there is no notion
+of task dependencies for task_group.
+
+The first extension is to expand the semantics and usable lifetime of
+`task_handle` so that remains valid after it is passed to run and it can 
+represent tasks in any state, including submitted, executing, and completed 
+tasks. Similarly, a `task_handle` in the submitted state may represent a task
```

Comment:
I think it will be necessary to specify in more detail what that means for a task handle to be valid and represent a task in a certain state. As a C++ object, it is valid until it is destroyed, and nothing changes in this regard I guess. But what you can and cannot do with a task handle might depend on the current state of the associated task.